File: BIG_QA_Solution/ScriptGenerator/scripts/templates/selenium_java_cucumber/src/test/java/stepDefinitions/LoginSteps.java
package stepDefinitions;

import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import org.junit.jupiter.api.Assertions;
import org.openqa.selenium.WebDriver;
import pageObjects.LoginPage;
import utils.ConfigReader;
import utils.DriverFactory;

/**
 * Step definitions for loginFeature.feature. Lives in the `stepDefinitions` glue package; the
 * driver is owned by hooks/Hooks.java, so we only fetch it here. Shipped sample that runs green
 * out of the box — replace with real steps when you write genuine tests.
 */
public class LoginSteps {

    private final WebDriver driver = DriverFactory.getDriver();
    private final LoginPage loginPage = new LoginPage(driver);

    @Given("I launch the application")
    public void i_launch_the_application() {
        loginPage.open(ConfigReader.getAppUrl());
    }

    @When("I enter valid Username and Password")
    public void i_enter_valid_username_and_password() {
        loginPage.login(
                ConfigReader.getOrDefault("USER", "standard_user"),
                ConfigReader.getOrDefault("PASSWORD", "secret_sauce"));
    }

    @When("I click the login button")
    public void i_click_the_login_button() {
        loginPage.submit();
    }

    @Then("I should be redirected to the homepage")
    public void i_should_be_redirected_to_the_homepage() {
        Assertions.assertTrue(loginPage.isLoaded(),
                "Expected the application page to load after login, but it did not. URL: " + driver.getCurrentUrl());
    }
}

File: BIG_QA_Solution/ScriptGenerator/scripts/templates/selenium_csharp_reqnroll/StepDefinitions/LoginSteps.cs
using NUnit.Framework;
using OpenQA.Selenium;
using Reqnroll;
using SeleniumReqnrollTests.PageObjects;
using SeleniumReqnrollTests.Utils;

namespace SeleniumReqnrollTests.StepDefinitions;

/// <summary>
/// Step definitions for LoginFeature.feature. The IWebDriver is created by Hooks and injected here
/// via Reqnroll's context injection (constructor parameter). Shipped sample that runs green out of
/// the box — replace with real steps when you write genuine tests.
/// </summary>
[Binding]
public class LoginSteps
{
    private readonly IWebDriver _driver;
    private readonly LoginPage _loginPage;

    public LoginSteps(IWebDriver driver)
    {
        _driver = driver;
        _loginPage = new LoginPage(driver);
    }

    [Given("I launch the application")]
    public void GivenILaunchTheApplication()
    {
        _loginPage.Open(ConfigReader.GetAppUrl());
    }

    [When("I enter valid Username and Password")]
    public void WhenIEnterValidUsernameAndPassword()
    {
        _loginPage.Login(
            ConfigReader.GetOrDefault("USER", "standard_user"),
            ConfigReader.GetOrDefault("PASSWORD", "secret_sauce"));
    }

    [When("I click the login button")]
    public void WhenIClickTheLoginButton()
    {
        _loginPage.Submit();
    }

    [Then("I should be redirected to the homepage")]
    public void ThenIShouldBeRedirectedToTheHomepage()
    {
        Assert.That(
            _loginPage.IsLoaded(),
            Is.True,
            $"Expected the application page to load after login, but it did not. URL: {_driver.Url}");
    }
}
